### ISSUE TITLE
`exa -l` error on specific directory

### ISSUE THREAD

**Issue by dontdieych** [ogham/exa#710]
```
a@a ~/d/AutoHotkey_2.0-a119-179d27fd> /bin/ls -alR
.:
total 3784
drwxr-xr-x  3 a a    4096 Aug  3 22:19 .
drwxrwxrwx 29 a a    4096 Aug  3 22:14 ..
-rw-r--r--  1 a a     170 Aug  3 22:17 ahk.ahk
-rw-r--r--  1 a a 1675927 Jul 26 19:25 AutoHotkey.chm
-rw-r--r--  1 a a  964096 Jul 26 19:24 AutoHotkeyU32.exe
-rw-r--r--  1 a a 1190400 Jul 26 19:24 AutoHotkeyU64.exe
drwxr-xr-x  2 a a    4096 Jan  1  1970 Compiler
-rw-r--r--  1 a a   18120 Oct  2  2016 license.txt

./Compiler:
total 1992
drwxr-xr-x 2 a a    4096 Jan  1  1970  .
drwxr-xr-x 3 a a    4096 Aug  3 22:19  ..
-rw-r--r-- 1 a a  917504 Jul 26 19:24 'Unicode 32-bit.bin'
-rw-r--r-- 1 a a 1111552 Jul 26 19:24 'Unicode 64-bit.bin'


a@a ~/d/AutoHotkey_2.0-a119-179d27fd> exa
license.txt  Compiler  AutoHotkeyU64.exe  AutoHotkeyU32.exe  AutoHotkey.chm  ahk.ahk


a@a ~/d/AutoHotkey_2.0-a119-179d27fd> exa -l
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: SystemTimeError(32400s)', src/fs/file.rs:331:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
fish: Job 1, “command exa -r $argv” terminated by signal SIGABRT (Abort)


a@a ~/d/AutoHotkey_2.0-a119-179d27fd [SIGABRT]> pwd
/home/a/daoc/AutoHotkey_2.0-a119-179d27fd

```

**Comment by dontdieych** [comment]
```
a@a ~/d/AutoHotkey_2.0-a119-179d27fd [SIGABRT]> RUST_BACKTRACE=full /usr/bin/exa -l

thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: SystemTimeError(32400s)', src/fs/file.rs:331:9
stack backtrace:
   0:     0x556bfebfc9e8 - <unknown>
   1:     0x556bfeb81a3c - <unknown>
   2:     0x556bfebfc0c4 - <unknown>
   3:     0x556bfebfbafa - <unknown>
   4:     0x556bfebfb29a - <unknown>
   5:     0x556bfebfae7a - <unknown>
   6:     0x556bfeb80110 - <unknown>
   7:     0x556bfeb82992 - <unknown>
   8:     0x556bfebaf378 - <unknown>
   9:     0x556bfeb95b7e - <unknown>
  10:     0x556bfeb9dae7 - <unknown>
  11:     0x556bfebf6415 - <unknown>
  12:     0x556bfebf532c - <unknown>
  13:     0x556bfec06435 - <unknown>
  14:     0x7fe957d50422 - start_thread
  15:     0x7fe957c65bf3 - __GI___clone
  16:                0x0 - <unknown>
fish: “RUST_BACKTRACE=full /usr/bin/ex…” terminated by signal SIGABRT (Abort)

a@a ~/d/AutoHotkey_2.0-a119-179d27fd [SIGABRT]>

**Comment by dontdieych** [comment]
Forgot about env info.

Manjaro Linux
exa v0.9.0

Let me know if you need more.

**Comment by pashosh** [comment]
also here:
`exa -l
thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 0', /checkout/src/liballoc/vec.rs:1555:10
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Aborted (core dumped)`

i'm on gentoo and `exa v0.8.0`

**Comment by pashosh** [comment]
also here:
>exa -l
thread 'main' panicked at 'index out of bounds: the len is 0 but the index is 0', /checkout/src/liballoc/vec.rs:1555:10
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Aborted (core dumped)`

i'm on gentoo and `exa v0.8.0`

**Comment by ogham** [comment]
Yeah, lots of people are running into this with Hugo, which generates files with timestamps _way_ in the past! exa wasn't handling the old timestamp properly.

The fix for this has been merged (#670) and will go out with the next release.
